### PR TITLE
[REFACTOR] on-off-farm uses new models

### PR DESCRIPTION
**Pull request by DrogoNevets**

### CODE REVIEW

File: src/server/common/controller/question-page-controller.test.js [hunk 1]
```diff
@@ -247,6 +247,30 @@ describe('QuestionPageController', () => {
     expect(statusCode).toBe(statusCodes.ok)
   })
 
+  test('Should display an error and set next page appropriately', async () => {
```
Comment (joseluisgraa) [inline src/server/common/controller/question-page-controller.test.js:250]: use "it" to keep it consistent

File: src/server/common/helpers/not-implmented-error.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export class NotImplementedError extends Error {
```
Comment (joseluisgraa) [inline src/server/common/helpers/not-implmented-error.js:1]: Can we rename the file as it has a typo?

File: src/server/common/model/answer/on-off-farm.js [hunk 3]
```diff
@@ -45,6 +47,16 @@ export class OnOffFarm extends AnswerModel {
     }
   }
 
+  /** @param {AnswerErrors} errors */
+  static errorMessages(errors) {
```
Comment (joseluisgraa) [inline src/server/common/model/answer/on-off-farm.js:51]: Are we not supposed to use the static method defined already in the answer model?

File: src/server/common/model/answer/on-off-farm.test.js [hunk 4]
```diff
@@ -73,6 +73,17 @@ describe('#OnOffFarm.fromState', () => {
   })
 })
 
+describe('OnOffFarm.errorMessages', () => {
```
Comment (joseluisgraa) [inline src/server/common/model/answer/on-off-farm.test.js:76]: If we just use the static method from answer model we might want to also unit test it in the model only?

File: src/server/origin/on-off-farm/index.js [hunk 5]
```diff
@@ -1,36 +1,52 @@
-import {
-  onOffFarmGetController,
-  onOffFarmPostController
-} from './controller.js'
-
 /**
- * Sets up the routes used in the home page.
+ * Sets up the routes used in the cph number page.
```
Comment (joseluisgraa) [inline src/server/origin/on-off-farm/index.js:2]: on-off-farm not cph number

File: src/server/common/model/answer/on-off-farm.js (hunk 3 above)
Comment (DrogoNevets) [reply src/server/common/model/answer/on-off-farm.js:51]: no as validation might be slightly different on each answer (one field vs many) so base throws a "not Implemented" error

File: src/server/common/model/answer/on-off-farm.test.js (hunk 4 above)
Comment (DrogoNevets) [reply src/server/common/model/answer/on-off-farm.test.js:76]: see above

File: src/server/origin/on-off-farm/index.js [hunk 6]
```diff
@@ -1,36 +1,52 @@
...
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  nextPage(answer) {
+    if (answer.value === 'on') {
+      return exitPage
     }
+    return cphNumberPage
+  }
+
+  /** @param {AnswerErrors} errors */
+  errorMessages(errors) {
```
[the remark this answers is not in the document]
Comment (DrogoNevets) [reply src/server/origin/on-off-farm/index.js:35]: as above, its different for multi field vs single

**Pull request merged**